aptos-move/framework/aptos-framework/sources/Account.move

```diff
@@ -343,18 +343,95 @@ module AptosFramework::Account {
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    /// Basic account creation method.
+    /// Basic account creation methods.
     ///////////////////////////////////////////////////////////////////////////
 
     public(script) fun create_account(auth_key: address) {
-        let (signer, _) = create_account_internal(auth_key);
+        let signer = create_account_internal(auth_key);
         Coin::register<TestCoin>(&signer);
     }
 
+    /// A resource account is used to manage resources independent of an account managed by a user.
+    /// Some applications might include managing a Dao and the resources associated with the Dao or
+    /// a liquidity pool to create new liquidity pool coins without requiring the original account
+    /// owner to setup the resources. While one could theoretically manage a lot of this via
+    /// capabilities, the `move_to` semantics insist on a signer and cannot work with capabilities
+    /// due to Move requirements that `move_to` be executed on a resource within the same module
+    /// that defines that resource.
+    ///
+    /// As a small example:
+    /// ```
+    /// let (signer, cap) = create_resource_account(&source);
+    /// let lp = LiquidityPool { signer_cap: cap, ... };
+    /// move_to(&signer, lp);
+    /// ```
+    ///
+    /// Later on during a coin registration:
+    /// ```
+    /// public fun add_coin<X, Y>(lp: &LP, x: Coin<x>, y: Coin<y>) {
+    ///     if(!exists<LiquidityCoin<X, Y>(LP::Address(lp), LiquidityCoin<X, Y>)) {
+    ///         let mint, burn = Coin::initialize<LiquidityCoin<X, Y>>(...);
+    ///         move_to(&create_signer_with_capability(&lp.cap), LiquidityCoin<X, Y>{ mint, burn });
+    ///     }
+    ///     ...
+    /// }
+    /// ```
+    public fun create_resource_account(
```

Comment:
How to prevent one of the signer from transferring all resource out of this account since signer has the highest access to all resources under the account? 
I think you mention it would be a separate PR. But what would the solution look like?

Reply:
In this case, the module / resource that owns this signer defines where the signer can be used. So the security is up to the author of those contracts. Much like the security of a signer is up to the owner of that account. We can add some awareness checks, but ultimately it is a user facing problem.

Reply:
Our current coin and token contracts allow the signer to transfer the resource. Does it mean we need to modify existing contract functions so that these functions verify the capability instead of signer?

Reply:
nah, this is more for a shared account concept via a dao or something other module / resource sharing.